### PR TITLE
Sg/update msys2 version

### PR DESCRIPTION
**Pull request by SSE4**
new PR as #135 appears to be inactive (author didn't merge fixes)

Specify library name and version:  **msys2/20190524**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [n/a - copied existing conanfile] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

The previous version of this recipe was quite outdated and I would like to upstream this bump to the latest published msys2 build's version. (This is a mere copy of the previous version, with an adjustment of the URLs and checksums)

### CODE REVIEW

**Comment by uilianries** [comment]
I'm curious why CCI didn't complained about this PR. You are changing 2 versions, removing `20161025` and adding `all`.

**Comment by danimtb** [comment]
The service does not complain about versions of the same package, but it does if there are changes in more than one different packages regardless of their versions. So this PR seems fine

**Comment by uilianries** [comment]
@danimtb I see, thanks!

File: recipes/msys2/all/conanfile.py [hunk 1]
```diff
@@ -33,12 +30,18 @@ def source(self):
...
         pass
 
-    def build(self):
-        arch = str(self.settings.arch_build)
-        tools.download(**self.conan_data["sources"][self.version][arch])
-        filename = self.conan_data["sources"][self.version][arch]["filename"]
-        sha256 = self.conan_data["checksum"][self.version][arch]["sha256"]
+    def _download(self, url, sha256):
+        from six.moves.urllib.parse import urlparse
+        filename = os.path.basename(urlparse(url).path)
```
Comment (danimtb) [inline recipes/msys2/all/conanfile.py]:
```suggestion
        filename = os.path.basename(urlparse(url).path)
```

Comment (danimtb) [inline recipes/msys2/all/conanfile.py]: Can we make this more explicit? the import and getting the name that way seems complex. Is it possible to compose the file name?

Comment (SSE4) [reply recipes/msys2/all/conanfile.py]: I am not sure I understand. suggestions are welcomed. here I just extract path component from the URL, e.g. `http://example.com/some/file.tgz?param=value` will result in `some/file.tgz`. then I just get filename itself - `file.tgz`. do you want something different?

Comment (danimtb) [reply recipes/msys2/all/conanfile.py]: Nevermind. I thought this was the source method

**Pull request merged**